xmodule/js/src/video/036_video_social_sharing.js

```diff
@@ -0,0 +1,111 @@
+(function(define) {
+    'use strict';
+    // VideoSocialSharingHandler module.
+    define(
+        'video/036_video_social_sharing.js', ['underscore', 'gettext'],
+        function(_, gettext) {
+            var VideoSocialSharingHandler, SocialSharingSite, SimpleSocialSharingSite, facebook, linkedin, twitter;
+            /**
+     * Video Social Sharing control module.
+     * @exports video/036_video_social_sharing.js
+     * @constructor
+     * @param {jquery Element} element
+     * @param {Object} options
+     */
+            VideoSocialSharingHandler = function(element, options) {
+                if (!(this instanceof VideoSocialSharingHandler)) {
+                    return new VideoSocialSharingHandler(element, options);
+                }
+
+                _.bindAll(this, 'clickHandler');
+
+                this.container = element;
+
+                if (this.container.find('.wrapper-downloads .wrapper-social-share')) {
+                    this.initialize();
+                }
+
+                return false;
+            };
+
+            VideoSocialSharingHandler.prototype = {
+                // Initializes the module.
+                initialize: function() {
+                    this.el = this.container.find('.wrapper-social-share');
+                    this.el.on('click', '.btn-link', this.clickHandler);
+                    this.baseVideoUrl = this.el.data('url');
+                    this.course_id = this.container.data('courseId');
+                    this.block_id = this.container.data('blockId')
+                    this.socialSharingSites = this.getSocialSharingSites()
+                },
+
+                clickHandler: function(event) {
+                    var self = this;
+                    event.preventDefault();
+                    var source = $(event.currentTarget).data('source')
+                    var utmQuery = $.param({
+                        utm_source: source,
+                        utm_medium: 'social',
+                        utm_campaign: 'social-share-exp',
+                    });
+                    var sharedVideoUrl = encodeURIComponent(self.baseVideoUrl + "?" + utmQuery);
+                    var socialShareSite = self.socialSharingSites[source];
+                    var socialMediaShareLinkUrl = socialShareSite.generateShareUrl(sharedVideoUrl);
+                    window.open(
+                        socialMediaShareLinkUrl,
+                        'targetWindow',
+                        'toolbar=no,location=0,status=no,menubar=no,scrollbars=yes,resizable=yes,width=600,height=400'
+                    );
+                    self.sendAnalyticsEvent(source);
+                },
+
+                getSocialSharingSites: function() {
```

Comment:
This is styled after a similar thing in `video/10_commands.js` that I thought was pretty neat